### PR TITLE
[PVR] Fix PVR not working after using kodi login screen (trac#16813)

### PR DESCRIPTION
**Pull request by ksooo**
Problem reported in trac http://trac.kodi.tv/ticket/16813 and the forum http://forum.kodi.tv/showthread.php?tid=282907&pid=2378582#pid2378582

Cause: Regression introduced by PVRManager refactoring for Krypton.
- first commit is basically preparation for the actual bug fix, to be able to do it in clean way
- second commit is the fix
- third commit fixes a race/deadlock that showed up after the login screen bug was fixed

@Jalle19 mind taking a look.

### CODE REVIEW

**Review by Jalle19 — APPROVED** [review]
Trying this new review feature. +1

**Comment by ksooo** [comment]
jenkins build this please

File: xbmc/pvr/windows/GUIWindowPVRBase.cpp [hunk 1]
```diff
@@ -1047,3 +1053,29 @@ bool CGUIWindowPVRBase::ConfirmStopRecording(const CPVRTimerInfoTagPtr &timer)
...
+{
+  if (!m_progressHandle)
+  {
+    CGUIDialogExtendedProgressBar *loadingProgressDialog = dynamic_cast<CGUIDialogExtendedProgressBar *>(g_windowManager.GetWindow(WINDOW_DIALOG_EXT_PROGRESS));
+    if (!loadingProgressDialog)
+    {
+      CLog::Log(LOGERROR, "CGUIWindowPVRBase - %s - unable to get WINDOW_DIALOG_EXT_PROGRESS!", __FUNCTION__);
+      return;
+    }
+    m_progressHandle = loadingProgressDialog->GetHandle(g_localizeStrings.Get(19235)); // PVR manager is starting up
```
Comment (Jalle19) [inline xbmc/pvr/windows/GUIWindowPVRBase.cpp]: Is `GetHandle()` guaranteed to not return null?

Comment (ksooo) [inline xbmc/pvr/windows/GUIWindowPVRBase.cpp]: yep, it is

**Pull request merged**